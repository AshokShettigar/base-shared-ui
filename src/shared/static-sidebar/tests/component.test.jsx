import React from 'react';
import { shallow } from 'enzyme';

import StaticSideBarContainer from '../component';

const defaultProps = {
    isExpand: true,
    title: 'default title',

    onCollapse: jest.fn(),
    onExpand: jest.fn(),
    children: <div>Children</div>
};

describe('Static side bar component', () => {
    it('should render component when its open', () => {
        const wrapper = shallow(<StaticSideBarContainer {...defaultProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component when its closed', () => {
        const wrapper = shallow(<StaticSideBarContainer {...defaultProps} isExpand={false} />);

        expect(wrapper).toMatchSnapshot();
    });
});
