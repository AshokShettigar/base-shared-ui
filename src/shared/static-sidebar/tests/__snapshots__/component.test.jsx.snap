// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static side bar component should render component when its closed 1`] = `
<div
  className="collapsed"
  data-component="StaticSidebar"
>
  <div
    className="header"
  >
    <WithRender
      className="backIcon"
      glyph="caret-left"
      hidden={false}
      onClick={[Function]}
      rendered={true}
      size="large"
    />
    <span
      className="title"
      hidden={true}
    >
      default title
    </span>
  </div>
  <div
    className="contentCollapsed content"
  />
</div>
`;

exports[`Static side bar component should render component when its open 1`] = `
<div
  className="expanded"
  data-component="StaticSidebar"
>
  <div
    className="header"
  >
    <span
      className="title"
      hidden={false}
    >
      default title
    </span>
    <WithRender
      className="close"
      glyph="close"
      hidden={false}
      onClick={[Function]}
      rendered={true}
      size="large"
    />
  </div>
  <div
    className="content"
  >
    <div>
      Children
    </div>
  </div>
</div>
`;
